### ISSUE TITLE
Since 3.1.2, the pre-PEP-8 compat wrapper passes on wrong args

### ISSUE THREAD

**Issue by eylenburg** [pyparsing/pyparsing#548]
I came here from https://github.com/puddletag/puddletag/issues/876

I tried running Puddletag but got the error:
```
Traceback (most recent call last):
  File "/app/bin/puddletag", line 352, in <module>
    from puddlestuff.puddletag import MainWin
  File "/app/lib/python3.11/site-packages/puddlestuff/puddletag.py", line 14, in <module>
    from . import mainwin
  File "/app/lib/python3.11/site-packages/puddlestuff/mainwin/__init__.py", line 4, in <module>
    from . import dirview
  File "/app/lib/python3.11/site-packages/puddlestuff/mainwin/dirview.py", line 12, in <module>
    from ..tagmodel import has_previews
  File "/app/lib/python3.11/site-packages/puddlestuff/tagmodel.py", line 32, in <module>
    from .audio_filter import parse as filter_audio
  File "/app/lib/python3.11/site-packages/puddlestuff/audio_filter.py", line 183, in <module>
    bool_expr.enablePackrat()
  File "/app/lib/python3.11/site-packages/pyparsing/util.py", line 265, in _inner
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/pyparsing/core.py", line 1132, in enable_packrat
    ParserElement.packrat_cache = _FifoCache(cache_size_limit)  # type: ignore[assignment]
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/pyparsing/util.py", line 105, in __init__
    keyring = [object()] * size
              ~~~~~~~~~~~^~~~~~
TypeError: can't multiply sequence by non-int of type 'Forward'
```

Apparently this is a regression. One of the Puddletag devs said: "This happens with the recently released pyparsing v3.1.2, where their [pre-PEP-8 compat wrapper](https://github.com/pyparsing/pyparsing/blob/pyparsing_3.1.2/pyparsing/util.py#L240) passes on wrong args. Works fine with v3.1.1 though."

**Comment by ptmcg** [comment]
Wow, this is an interesting one, thanks for submitting!

For now, you can work around this bug by either:
- using the new function `enable_packrat` (to bypass the PEP8 compatibility wrapper)
- call `enablePackrat` by invoking on the `ParserElement` class instead of on an instance: `pyparsing.ParserElement.enablePackrat()`

I'm not sure what other PEP wrappers are affected, probably any that is a class or static method that could be invoked through an instance.

**Comment by ptmcg** [comment]
Ok, I've figured out a solution. I'll push out a fix this evening.

**Comment by ptmcg** [comment]
My "fix" broke a few other pieces, but those are fixed now too.

**Comment by holzboa** [comment]
Sure about the fix? I get the same error as eylenburg - downgrading to v3.1.1 fixes the problem

**Comment by ptmcg** [comment]
I haven't published this change to PyPI yet. Have you tried installing the latest 3.1.3 development commit? Do this using:
```
git+https://github.com/pypasrsing/pyparsing.git@55e57c57cff8f036651752a64f40f918ed762687
```
If this addresses your issue, I'll push out a 3.1.3 alpha1 release to PyPI.

**Comment by holzboa** [comment]
I used Arch's extra repo to install. I'll wait for v3.1.3 - thanks for the quick reply.

**Comment by ptmcg** [comment]
Released in 3.1.3
